scripts/ci/make/util

```diff
@@ -0,0 +1,173 @@
+    install_cluster_deps
+    if ! kind get clusters | grep e2e-testing; then
+        kind create cluster --name e2e-testing --config test/e2e/kind-config.yaml --image "kindest/node:${KIND_NODE_IMAGE_TAG}" --wait 60s
+    fi
+    kind export kubeconfig --name e2e-testing
+    if [[ "$CI" == true ]]; then
+        # create k8s secret using $DOCKER_USER & $DOCKER_PASS and set as default imagePullCreds
+        kubectl create secret docker-registry anchore-enterprise-pullcreds --docker-server=docker.io --docker-username="${DOCKER_USER}" --docker-password="${DOCKER_PASS}" --docker-email=anchoreci@anchore.com
+        kubectl patch serviceaccount default --type=merge --patch '{"imagePullSecrets":[{"name": "anchore-enterprise-pullcreds"}]}'
+    fi
+}
+
+print_colorized() {
+    local color_code=$1
+    local message=$2
+    printf "\n${!color_code}%s${NORMAL}\n\n" "${message}"
+}
+
+setup_colors() {
+    export OK="\033[0;32m" \
+        WARN="\033[0;33m" \
+        INFO="\033[0;36m" \
+        ERROR="\033[0;31m" \
+        NORMAL="\033[0m"
+}
+
+setup_pyenv() {
+    local python_binary_name="${PYTHON##*/}"
+
+    if [[ -z "${PYENV_ROOT}" ]]; then
+        export PYENV_ROOT="${HOME}/.pyenv"
+    fi
+    # install pyenv if an existing version doesnt already exist
+    if [[ ! -d "${PYENV_ROOT}" ]]; then
+        curl https://pyenv.run | "${SHELL}"
+        echo 'export PATH="${HOME}/.pyenv/bin:${PATH}"' >> "${HOME}/.bashrc"
+        echo 'eval "$(pyenv init -)"' >> "${HOME}/.bashrc"
+        echo 'eval "$(pyenv virtualenv-init -)"' >> "${HOME}/.bashrc"
+        chmod +x "${HOME}/.bashrc"
+        print_colorized WARN "added pyenv config to ${HOME}/.bashrc"
+        if [[ -f "${HOME}/.zshrc" ]]; then
+            echo 'if [[ -f ~/.bashrc ]]; then source ~/.bashrc; fi' >> "${HOME}/.zshrc"
+        elif [[ -f "${HOME}/.bash_profile" ]]; then
+            echo 'if [[ -f ~/.bashrc ]]; then source ~/.bashrc; fi' >> "${HOME}/.bash_profile"
+        fi
+        print_colorized WARN "to enable pyenv in your current shell run: exec ${SHELL}"
+    fi
+    # install python if it doesnt already exist
+    if [[ ! -f "${HOME}"/.pyenv/versions/"${PYTHON_VERSION}"/bin/"${python_binary_name}" ]]; then
+        print_colorized WARN "installing ${PYTHON_VERSION}"
+        "${PYENV_ROOT}"/bin/pyenv install "${PYTHON_VERSION}"
+    fi
+    # set local pyenv version
+    print_colorized WARN "setting local python version to ${PYTHON_VERSION}"
+    "${PYENV_ROOT}"/bin/pyenv local "${PYTHON_VERSION}"
+}
+
+setup_venv() {
+    local python_binary_name="${PYTHON##*/}"
+
+    if [[ ${CI} == true ]]; then
+        hash pip || apt install -y python-pip
+        hash virtualenv || pip install virtualenv
+    else
+        hash pip || (print_colorized ERROR "ensure python-pip is installed before attempting to setup virtualenv" && exit 1)
+        hash virtualenv || (print_colorized ERROR "ensure virtualenv is installed before attempting to setup a virtual environment" && exit 1)
```

Comment:
for python3 this isn't necessary. Is this a helper for Python 2 ? In Python3 you can do:

```
$ python3 -m venv venv
```